docs/cpu_templates/cpu-templates.md

```diff
@@ -0,0 +1,190 @@
+# CPU templates
+
+Firecracker allows users to customise how the vCPUs are represented
+to the guest software by changing the following configuration:
+
+- CPUID (x86_64 only)
+- MSRs (Model Specific Registers, x86_64 only)
+- ARM registers (aarch64 only)
+
+A combination of the changes to the entities above is called a CPU template.
+
+The functionality can be used when a user wants to mask a feature from
+the guest. One of real world use cases for this is representing a heterogeneous
+fleet (a fleet consisting of multiple CPU models) as a homogeneous fleet,
+so the guests will experience a consistent feature set supported by the host.
+
+> **Note**
+Representing one CPU vendor as another CPU vendor is not supported.
+
+> **Note**
+CPU templates is not a security feature. It does not improve
+the guest's security posture in any way. It is only supposed to modify
+indications to the guest software on what hardware features are available.
```

Comment:
nit: We say it `does not improve the guest's security posture` but we do not say `it does not degrade the guest's security posture` we can rule out both by using `affect`.
```suggestion
CPU templates is not a security feature. It does not affect
the guest's security posture in any way. It is only supposed to modify
indications to the guest software on what hardware features are available.
```

Reply:
Okay, it may just be me, but maybe this can be re-worded like:

> CPU templates is not inherently a security feature without specific configuration it does not affect a guest's security posture. When specifically configured it can however affect the security of a guest.

Reply:
I think I found a way to express what I actually meant. Updated.

Reply:
Okay lgtm